R/batchmark.R

```diff
@@ -0,0 +1,175 @@
+#' each job is a full run of inner resampling, which can be parallelized in a second step
+#' with \pkg{ParallelMap}. For details on the usage and support backends have
+#' a look at the batchtools tutorial page:
+#' \url{https://github.com/mllg/batchtools}.
+#'
+#' The general workflow with \code{batchmark} looks like that:
+#' \enumerate{
+#' \item{1.}{\code{batchtools::createExperimentRegistry()}}
+#' \item{2.}{\code{batchmark(...)}}
+#' \item{3.}{\code{batchtools:submitJobs()}}
+#' \item{4.}{wait until the jobs are finished...}
+#' \item{5.}{\code{reduceBatchtoolsResult()}}
+#' }
+#' @inheritParams benchmark
+#' @param data.ids [\code{integer}]\cr
+#'   Dataset IDs to download from Open-ML. Default is none.
+#' @param resamplings [(list of) \code{\link{ResampleDesc}}]\cr
+#'   Resampling strategy for each tasks.
+#'   If only one is provided, it will be replicated to match the number of tasks.
+#'   If missing, a 10-fold cross validation is used.
+#' @param reg [\code{\link[batchtools]{Registry}}]\cr
+#'   Registry, created by \code{\link[batchtools]{makeExperimentRegistry}}. If not explicitly passed,
+#'   uses the last created registry.
+#' @return [\code{data.table}]. Generated job ids are stored in the column \dQuote{job.id}.
+#' @export
+#' @family benchmark
+batchmark = function(learners, tasks, data.ids, resamplings, measures, models = TRUE, reg = batchtools::getDefaultRegistry()) {
+
+  requirePackages("batchtools", why = "batchmark", default.method = "load")
+
+  learners = ensureBenchmarkLearners(learners)
+  learner.ids = vcapply(learners, getLearnerId)
+  names(learners) = learner.ids
+
+  if (!missing(tasks)) {
+    tasks = ensureBenchmarkTasks(tasks)
+    task.ids = vcapply(tasks, getTaskId)
+    names(tasks) = task.ids
+  }
+  if (!missing(data.ids)) {
+    requirePackages("OpenML", why = "OpenML data ids given", default.method = "load")
+    OpenML::populateOMLCache(data.ids = data.ids)
+    did.tasks = as.list(data.ids)
+    names(did.tasks) = data.ids
+    tasks = if (missing(tasks))
+      did.tasks
+    else
+      c(tasks, did.tasks)
+    task.ids = names(tasks)
+  }
+  if (missing(tasks))
+    stop("At least one task or OpenML data ID has to be given")
+
+  resamplings = ensureBenchmarkResamplings(resamplings, tasks)
+  names(resamplings) = task.ids
+
+  measures = ensureBenchmarkMeasures(measures, tasks)
+
+  assertFlag(models)
+
+  reg$packages = union(reg$packages, "mlr")
+
+  # generate problems
+  pdes = Map(function(id, task, rdesc, seed) {
+    batchtools::addProblem(id, data = list(rdesc = rdesc, task = task, measures = measures, learners = learners), fun = resample.fun, seed = seed, reg = reg)
+    data.table(i = seq_len(rdesc$iters))
+  }, id = task.ids, task = tasks, rdesc = resamplings, seed = reg$seed + seq_along(tasks))
+
+  # generate algos
+  ades = Map(function(id, learner) {
+    apply.fun = getAlgoFun(learner, measures, models)
+    batchtools::addAlgorithm(id, apply.fun, reg = reg)
+    data.table()
```

Comment:
Yep, `addExperiments` expects a named list of `data.frame`/`data.table`. This is an easy way to construct it here.